File: go/wordcount.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"regexp"
	"sort"
	"strings"
)

func main() {
	counts := make(map[string]int)
	scanner := bufio.NewScanner(os.Stdin)
	r := regexp.MustCompile(`[a-z\']+`)
	for scanner.Scan() {
		line := strings.ToLower(scanner.Text())
		for _, word := range r.FindAllString(line, -1) {
			counts[word] += 1
		}
	}
	report(counts)
}

func report(counts map[string]int) {
	var words []string
	for word := range counts {
		words = append(words, word)
	}
	sort.Strings(words)
	for _, word := range words {
		fmt.Println(word, counts[word])
	}
}

File: perl/wordcount.pl
use v5.40;
use utf8;
use open qw(:std :utf8);

my %word_count;

while (my $line = <>) {
    chomp $line;

    my @words = ($line =~ /[\p{L}'’]+/g);

    foreach my $word (@words) {
        $word_count{lc $word}++;
    }
}

foreach my $word (sort keys %word_count) {
    say "$word $word_count{$word}";
}
